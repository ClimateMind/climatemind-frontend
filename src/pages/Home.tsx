import React from 'react';
import { useHistory } from 'react-router-dom';
import { Grid, Box, Typography, Button, makeStyles } from '@material-ui/core';
import { ReactComponent as Logo } from '../assets/cm-logo-home.svg';
import ROUTES from '../components/Router/RouteConfig';

const styles = makeStyles({
  root: {
    backgroundColor: '#82EFC5',
    minHeight: '100vh',
  },
  typography: {
    wordSpacing: '100vw',
  },
});

const Home: React.FC<{}> = () => {
  const classes = styles();
  const history = useHistory();

  return (
    <Grid
      container
      data-testid="Home"
      direction="column"
      alignItems="center"
      alignContent="center"
      className={classes.root}
    >
      <Grid item>
        <Box mt={25} mb={3}>
          <Logo data-testid="climate-mind-logo" />
        </Box>
      </Grid>

      <Grid item>
        <Box mt={22}>
          <Typography
            variant="h4"
            align="center"
            className={classes.typography}
          >
            Powering climate conversations
          </Typography>
        </Box>
      </Grid>

      <Grid item xs={12}>
        <Box mt={6} mb={6}>
          <Button
            variant="contained"
            color="primary"
            disableElevation
            onClick={() => history.push(ROUTES.ROUTE_QUIZHOME)}
          >
            Get Started
          </Button>
        </Box>
      </Grid>
    </Grid>
  );
};

export default Home;
