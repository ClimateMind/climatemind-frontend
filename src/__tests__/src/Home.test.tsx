import React from 'react';
import { render } from '@testing-library/react';
import Home from '../../pages/Home';

describe('Home page', () => {
  it('has a get started button', () => {
    const { getByRole } = render(<Home />);
    expect(getByRole('button', { name: /get started/i })).toBeInTheDocument();
  });
  it('displays the app logo', () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId('climate-mind-logo')).toBeInTheDocument();
  });
  it('shows Powering climate conversations', () => {
    const { getByText } = render(<Home />);
    expect(getByText(/Powering climate conversations/i)).toBeInTheDocument();
  });
});
