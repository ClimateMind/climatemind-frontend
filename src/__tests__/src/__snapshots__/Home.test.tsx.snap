// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Matchs   the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-root-16 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center"
  data-testid="Home"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiBox-root MuiBox-root-18"
    >
      <svg
        data-testid="climate-mind-logo"
      >
        cm-logo-home.svg
      </svg>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiBox-root MuiBox-root-19"
    >
      <h4
        class="MuiTypography-root makeStyles-typography-17 MuiTypography-h4 MuiTypography-alignCenter"
      >
        Powering climate conversations
      </h4>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiBox-root MuiBox-root-20"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Get Started
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
